packages/compass-app-stores/src/provider.tsx

```diff
@@ -0,0 +1,188 @@
+import {
+  dataServiceLocator,
+  useConnectionInfo,
+} from '@mongodb-js/compass-connections/provider';
+import {
+  createServiceLocator,
+  createServiceProvider,
+} from 'hadron-app-registry';
+import type { MongoDBInstance } from 'mongodb-instance-model';
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { type MongoDBInstancesManager } from './instances-manager';
+import toNS from 'mongodb-ns';
+import type Collection from 'mongodb-collection-model';
+import type Database from 'mongodb-database-model';
+
+export {
+  MongoDBInstancesManagerEvents,
+  type MongoDBInstancesManager,
+} from './instances-manager';
+
+export const MongoDBInstancesManagerContext =
+  createContext<MongoDBInstancesManager | null>(null);
+
+export const MongoDBInstancesManagerProvider =
+  MongoDBInstancesManagerContext.Provider;
+
+export const mongoDBInstancesManagerLocator = createServiceLocator(
+  function useMongoDBInstanceManager(): MongoDBInstancesManager {
+    const instancesManager = useContext(MongoDBInstancesManagerContext);
+    if (!instancesManager) {
+      throw new Error(
+        'No MongoDBInstancesManager available in this context, provider was not setup correctly'
+      );
+    }
+    return instancesManager;
+  },
+  'mongoDBInstancesManagerLocator'
+);
+
+export const mongoDBInstanceLocator = createServiceLocator(
+  function useMongoDBInstance(): MongoDBInstance {
+    const connectionInfo = useConnectionInfo();
+    const instancesManager = mongoDBInstancesManagerLocator();
+    const instance = instancesManager.getMongoDBInstanceForConnection(
+      connectionInfo.id
+    );
+    if (!instance) {
+      throw new Error('No MongoDBInstance available in this context');
+    }
+    return instance;
+  },
+  'mongoDBInstanceLocator'
+);
+
+const NamespaceModelContext = React.createContext<Database | Collection | null>(
```

Comment:
This file was renamed, but git diff thinks it was removed and added. New code starts from here, everything else is existing providers and locators for the instance model